### ISSUE TITLE
Weird bug with org-mode 9.7, poly-org and doom-docs

### ISSUE THREAD

**Issue by maikol-solis** [doomemacs/doomemacs#7913]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.
- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

When I got to source block with some elisp code, the mode should be Emacs-lisp and the cursor should stay in the same position. 

### Current behavior

Instead, a weird glitch occurs, and the cursor starts moving to the bottom without any control. I recorded the behavior:

https://imgur.com/a/PcY7PV9 or
https://i.imgur.com/Q19mHi1.mp4

And the backtrace

```elisp 
Debugger entered--Lisp error: (wrong-type-argument integer-or-marker-p nil)
  org-display-user-inline-images()
  apply(org-display-user-inline-images nil)
  org-display-inline-images()
  (if (local-variable-p 'org-startup-with-inline-images) nil (setq org-display-remote-inline-images 'cache) (org-display-inline-images))
  (if org-inhibit-startup nil (doom/reload-docs) (if (local-variable-p 'org-startup-with-inline-images) nil (setq org-display-remote-inline-images 'cache) (org-display-inline-images)) (if (local-variable-p 'org-startup-indented) nil (org-indent-mode 1)) (if (local-variable-p 'org-startup-numerated) nil (if (and (boundp 'org-num-mode) org-num-mode) (progn (org-num-mode -1))) (org-num-mode 1)) (if (local-variable-p 'org-startup-folded) nil (let ((org-startup-folded 'content) org-cycle-hide-drawer-startup) (org-set-startup-visibility))))
  (let ((gc-cons-threshold most-positive-fixnum) (gc-cons-percentage 1.0)) (require 'org-id) (require 'ob) (progn (set (make-local-variable 'org-id-link-to-org-use-id) t) (set (make-local-variable 'org-id-method) 'uuid) (set (make-local-variable 'org-id-track-globally) t) (set (make-local-variable 'org-id-locations-file) doom-docs--id-location-file) (set (make-local-variable 'org-id-locations) doom-docs--id-locations) (set (make-local-variable 'org-id-files) doom-docs--id-files) (set (make-local-variable 'org-num-max-level) 3) (set (make-local-variable 'org-footnote-define-inline) nil) (set (make-local-variable 'org-footnote-auto-label) t) (set (make-local-variable 'org-footnote-auto-adjust) t) (set (make-local-variable 'org-footnote-section) nil) (set (make-local-variable 'wgrep-change-readonly-file) t) (set (make-local-variable 'org-link-abbrev-alist-local) (append org-link-abbrev-alist-local doom-docs-link-alist)) (set (make-local-variable 'org-babel-default-header-args) (append '((:eval . "no") (:tangle . "no")) org-babel-default-header-args)) (set (make-local-variable 'save-place-ignore-files-regexp) ".")) (if org-inhibit-startup nil (doom/reload-docs) (if (local-variable-p 'org-startup-with-inline-images) nil (setq org-display-remote-inline-images 'cache) (org-display-inline-images)) (if (local-variable-p 'org-startup-indented) nil (org-indent-mode 1)) (if (local-variable-p 'org-startup-numerated) nil (if (and (boundp 'org-num-mode) org-num-mode) (progn (org-num-mode -1))) (org-num-mode 1)) (if (local-variable-p 'org-startup-folded) nil (let ((org-startup-folded 'content) org-cycle-hide-drawer-startup) (org-set-startup-visibility)))) (add-hook 'read-only-mode-hook #'doom-docs--toggle-read-only-h nil 'local))
  (let ((delay-mode-hooks t)) (org-mode) (setq major-mode 'doom-docs-org-mode) (setq mode-name "Doom Docs") (progn (if (get 'org-mode 'mode-class) (put 'doom-docs-org-mode 'mode-class (get 'org-mode 'mode-class))) (if (keymap-parent doom-docs-org-mode-map) nil (set-keymap-parent doom-docs-org-mode-map (current-local-map))) (let ((parent (char-table-parent doom-docs-org-mode-syntax-table))) (if (and parent (not (eq parent (standard-syntax-table)))) nil (set-char-table-parent doom-docs-org-mode-syntax-table (syntax-table)))) (if (or (abbrev-table-get doom-docs-org-mode-abbrev-table :parents) (eq doom-docs-org-mode-abbrev-table local-abbrev-table)) nil (abbrev-table-put doom-docs-org-mode-abbrev-table :parents (list local-abbrev-table)))) (use-local-map doom-docs-org-mode-map) (set-syntax-table doom-docs-org-mode-syntax-table) (setq local-abbrev-table doom-docs-org-mode-abbrev-table) (font-lock-add-keywords nil doom-docs-org-font-lock-keywords) (let ((gc-cons-threshold most-positive-fixnum) (gc-cons-percentage 1.0)) (require 'org-id) (require 'ob) (progn (set (make-local-variable 'org-id-link-to-org-use-id) t) (set (make-local-variable 'org-id-method) 'uuid) (set (make-local-variable 'org-id-track-globally) t) (set (make-local-variable 'org-id-locations-file) doom-docs--id-location-file) (set (make-local-variable 'org-id-locations) doom-docs--id-locations) (set (make-local-variable 'org-id-files) doom-docs--id-files) (set (make-local-variable 'org-num-max-level) 3) (set (make-local-variable 'org-footnote-define-inline) nil) (set (make-local-variable 'org-footnote-auto-label) t) (set (make-local-variable 'org-footnote-auto-adjust) t) (set (make-local-variable 'org-footnote-section) nil) (set (make-local-variable 'wgrep-change-readonly-file) t) (set (make-local-variable 'org-link-abbrev-alist-local) (append org-link-abbrev-alist-local doom-docs-link-alist)) (set (make-local-variable 'org-babel-default-header-args) (append '((:eval . "no") (:tangle . "no")) org-babel-default-header-args)) (set (make-local-variable 'save-place-ignore-files-regexp) ".")) (if org-inhibit-startup nil (doom/reload-docs) (if (local-variable-p 'org-startup-with-inline-images) nil (setq org-display-remote-inline-images 'cache) (org-display-inline-images)) (if (local-variable-p 'org-startup-indented) nil (org-indent-mode 1)) (if (local-variable-p 'org-startup-numerated) nil (if (and (boundp 'org-num-mode) org-num-mode) (progn (org-num-mode -1))) (org-num-mode 1)) (if (local-variable-p 'org-startup-folded) nil (let ((org-startup-folded 'content) org-cycle-hide-drawer-startup) (org-set-startup-visibility)))) (add-hook 'read-only-mode-hook #'doom-docs--toggle-read-only-h nil 'local)))
  (progn (make-local-variable 'delay-mode-hooks) (let ((delay-mode-hooks t)) (org-mode) (setq major-mode 'doom-docs-org-mode) (setq mode-name "Doom Docs") (progn (if (get 'org-mode 'mode-class) (put 'doom-docs-org-mode 'mode-class (get 'org-mode 'mode-class))) (if (keymap-parent doom-docs-org-mode-map) nil (set-keymap-parent doom-docs-org-mode-map (current-local-map))) (let ((parent (char-table-parent doom-docs-org-mode-syntax-table))) (if (and parent (not (eq parent ...))) nil (set-char-table-parent doom-docs-org-mode-syntax-table (syntax-table)))) (if (or (abbrev-table-get doom-docs-org-mode-abbrev-table :parents) (eq doom-docs-org-mode-abbrev-table local-abbrev-table)) nil (abbrev-table-put doom-docs-org-mode-abbrev-table :parents (list local-abbrev-table)))) (use-local-map doom-docs-org-mode-map) (set-syntax-table doom-docs-org-mode-syntax-table) (setq local-abbrev-table doom-docs-org-mode-abbrev-table) (font-lock-add-keywords nil doom-docs-org-font-lock-keywords) (let ((gc-cons-threshold most-positive-fixnum) (gc-cons-percentage 1.0)) (require 'org-id) (require 'ob) (progn (set (make-local-variable 'org-id-link-to-org-use-id) t) (set (make-local-variable 'org-id-method) 'uuid) (set (make-local-variable 'org-id-track-globally) t) (set (make-local-variable 'org-id-locations-file) doom-docs--id-location-file) (set (make-local-variable 'org-id-locations) doom-docs--id-locations) (set (make-local-variable 'org-id-files) doom-docs--id-files) (set (make-local-variable 'org-num-max-level) 3) (set (make-local-variable 'org-footnote-define-inline) nil) (set (make-local-variable 'org-footnote-auto-label) t) (set (make-local-variable 'org-footnote-auto-adjust) t) (set (make-local-variable 'org-footnote-section) nil) (set (make-local-variable 'wgrep-change-readonly-file) t) (set (make-local-variable 'org-link-abbrev-alist-local) (append org-link-abbrev-alist-local doom-docs-link-alist)) (set (make-local-variable 'org-babel-default-header-args) (append '(... ...) org-babel-default-header-args)) (set (make-local-variable 'save-place-ignore-files-regexp) ".")) (if org-inhibit-startup nil (doom/reload-docs) (if (local-variable-p 'org-startup-with-inline-images) nil (setq org-display-remote-inline-images 'cache) (org-display-inline-images)) (if (local-variable-p 'org-startup-indented) nil (org-indent-mode 1)) (if (local-variable-p 'org-startup-numerated) nil (if (and (boundp ...) org-num-mode) (progn (org-num-mode -1))) (org-num-mode 1)) (if (local-variable-p 'org-startup-folded) nil (let ((org-startup-folded ...) org-cycle-hide-drawer-startup) (org-set-startup-visibility)))) (add-hook 'read-only-mode-hook #'doom-docs--toggle-read-only-h nil 'local))))
  doom-docs-org-mode()
  #<subr hack-one-local-variable>(mode doom-docs-org)
  apply(#<subr hack-one-local-variable> (mode doom-docs-org))
  #f(advice TeX--call-minor-mode :after #<subr hack-one-local-variable>)(mode doom-docs-org)
  ad-Advice-hack-one-local-variable(#f(advice TeX--call-minor-mode :after #<subr hack-one-local-variable>) mode doom-docs-org)
  apply(ad-Advice-hack-one-local-variable #f(advice TeX--call-minor-mode :after #<subr hack-one-local-variable>) (mode doom-docs-org))
  hack-one-local-variable(mode doom-docs-org)
  hack-local-variables-apply()
  #<subr hack-local-variables>(no-mode)
  so-long--hack-local-variables(#<subr hack-local-variables> no-mode)
  apply(so-long--hack-local-variables #<subr hack-local-variables> no-mode)
  hack-local-variables(no-mode)
  run-mode-hooks(org-mode-hook)
  org-mode()
  pm--mode-setup(org-mode)
  #f(compiled-function (config) "Initialization of host buffers.\nRan by the polymode mode function." #<bytecode -0x1f74288d27651cb8>)(#<pm-polymode poly-org-polymode:>)
  apply(#f(compiled-function (config) "Initialization of host buffers.\nRan by the polymode mode function." #<bytecode -0x1f74288d27651cb8>) #<pm-polymode poly-org-polymode:> nil)
  pm-initialize(#<pm-polymode poly-org-polymode:>)
  poly-org-mode()
  set-auto-mode-0(poly-org-mode nil)
  set-auto-mode--apply-alist((("\\.[rR][mM][dD]\\.orig\\'" . poly-markdown+r-mode) ("\\.[rR]md" . poly-gfm+r-mode) ("\\.[pP][dD][fF]\\'" . pdf-view-mode) ("\\.org\\'" . poly-org-mode) ("Procfile\\'" . yaml-mode) ("/bspwmrc\\'" . sh-mode) ("\\.\\(?:zunit\\|env\\)\\'" . sh-mode) ("\\.bats\\'" . sh-mode) ("/Pipfile\\'" . conf-mode) ("[./]flake8\\'" . conf-mode) ("/README\\(?:\\.md\\)?\\'" . gfm-mode) ("\\.tex\\'" . LaTeX-mode) ("\\.js\\(?:on\\|[hl]int\\(?:rc\\)?\\)\\'" . json-mode) ("\\.[qQ][mM][dD]" . poly-quarto-mode) ("\\.[Jj][Mm][Dd]\\'" . ess-jags-mode) ("\\.[Jj][Oo][Gg]\\'" . ess-jags-mode) ("\\.[Rr]out" . R-transcript-mode) ("\\.Sout" . S-transcript-mode) ("\\.ado\\'" . STA-mode) ("\\.do\\'" . STA-mode) ("\\.lsp\\'" . XLS-mode) ("\\.hat\\'" . omegahat-mode) ("\\.omg\\'" . omegahat-mode) ("CITATION\\'" . R-mode) ("NAMESPACE\\'" . R-mode) ("\\.[rR]profile\\'" . R-mode) ("\\.[sS]nw\\'" . poly-noweb+r-mode) ("\\.[rR]\\'" . R-mode) ("\\.SSC\\'" . S-mode) ("\\.[Ss][s][c]\\'" . S-mode) ("\\.[qsS]\\'" . S-mode) ("/R/.*\\.q\\'" . R-mode) ("\\.sp\\'" . S-mode) ("\\.[Jj][Ll]\\'" . ess-julia-mode) ("\\.Cask\\'" . emacs-lisp-mode) ("\\.h\\'" . +cc-c-c++-objc-mode) ("\\.mm\\'" . objc-mode) ("\\.pdf\\'" . pdf-view-mode) ("\\.nw\\'" . poly-noweb-mode) ("\\.editorconfig\\'" . editorconfig-conf-mode) ("\\.tsv\\'" . tsv-mode) ("\\.[Cc][Ss][Vv]\\'" . csv-mode) ("\\.\\(e?ya?\\|ra\\)ml\\'" . yaml-mode) ("requirements\\.in" . pip-requirements-mode) ("requirements[^z-a]*\\.txt\\'" . pip-requirements-mode) ("\\.pip\\'" . pip-requirements-mode) ("\\.hva\\'" . LaTeX-mode) ("\\(?:\\(?:\\.\\(?:b\\(?:\\(?:abel\\|ower\\)rc\\)\\|json\\(?:l..." . json-mode) ("\\.qmd\\'" . poly-quarto-mode) ("\\.[Ss][Aa][Ss]\\'" . SAS-mode) ...) nil nil)
  #<subr set-auto-mode>()
  apply(#<subr set-auto-mode> nil)
  (progn (apply fn args))
  (unwind-protect (progn (apply fn args)) (when (buffer-live-p buf) (with-current-buffer buf (evil-change-state old-state))))
  (let ((evil-state evil-state) (evil-previous-state evil-previous-state) (evil-previous-state-alist (copy-tree evil-previous-state-alist)) (evil-next-state evil-next-state) (old-state evil-state) (inhibit-quit t) (buf (current-buffer))) (unwind-protect (progn (apply fn args)) (when (buffer-live-p buf) (with-current-buffer buf (evil-change-state old-state)))))
  (evil-save-state (apply fn args))
  (if evil-state (evil-save-state (apply fn args)) (apply fn args))
  +evil--persist-state-a(#<subr set-auto-mode>)
  apply(+evil--persist-state-a #<subr set-auto-mode> nil)
  #f(advice +evil--persist-state-a :around #<subr set-auto-mode>)()
  so-long--set-auto-mode(#f(advice +evil--persist-state-a :around #<subr set-auto-mode>))
  apply(so-long--set-auto-mode #f(advice +evil--persist-state-a :around #<subr set-auto-mode>) nil)
  #f(advice so-long--set-auto-mode :around #f(advice +evil--persist-state-a :around #<subr set-auto-mode>))()
  apply(#f(advice so-long--set-auto-mode :around #f(advice +evil--persist-state-a :around #<subr set-auto-mode>)) nil)
  set-auto-mode()
  normal-mode(t)
  #<subr after-find-file>(nil t nil nil nil)
  apply(#<subr after-find-file> (nil t))
  (progn (fset #'sit-for #'ignore) (apply fn args))
  (unwind-protect (progn (fset #'sit-for #'ignore) (apply fn args)) (fset #'sit-for old))
  (let* ((old (symbol-function #'sit-for))) (unwind-protect (progn (fset #'sit-for #'ignore) (apply fn args)) (fset #'sit-for old)))
  doom--shut-up-autosave-a(#<subr after-find-file> nil t)
  apply(doom--shut-up-autosave-a #<subr after-find-file> (nil t))
  #f(advice doom--shut-up-autosave-a :around #<subr after-find-file>)(nil t)
  apply(#f(advice doom--shut-up-autosave-a :around #<subr after-find-file>) (nil t))
  #f(advice chain-doom-first-buffer-hook-to-find-file-hook-h :before #f(advice doom--shut-up-autosave-a :around #<subr after-find-file>) ((depth . -101)))(nil t)
  apply(#f(advice chain-doom-first-buffer-hook-to-find-file-hook-h :before #f(advice doom--shut-up-autosave-a :around #<subr after-find-file>) ((depth . -101))) (nil t))
  after-find-file(nil t)
  find-file-noselect-1(#<buffer README.org> "~/.emacs.d/modules/completion/corfu/README.org" nil nil "~/.emacs.d/modules/completion/corfu/README.org" (336553232 16777223))
  find-file-noselect("~/.emacs.d/modules/completion/corfu/README.org" nil nil nil)
  find-file("~/.emacs.d/modules/completion/corfu/README.org")
  #<subr consult-recent-file>()
  apply(#<subr consult-recent-file> nil)
  consult-recent-file()
  funcall-interactively(consult-recent-file)
  command-execute(consult-recent-file)
```

### Steps to reproduce

1. Open Emacs, 
2. Have poly-org activated
3. Locate a file like `corfu/README.org` or `config.org`
4. go to a source block with elisp code
5. Glitch occurs. 

### System Information

https://pastebin.com/V6Na8szL

**Comment by hlissner** [comment]
`org-display-user-inline-images` is defined in `org-yt`. Try disabling the `org-yt` package and see if the issue goes away.

That said, I don't think poly-org is Doom's jurisdiction, since `doom-docs-mode` works just fine with `org-mode` alone.

**Comment by maikol-solis** [comment]
Hi Henrik!

I disabled `org-yt` and nothing changed. The problem was directly with `poly-org`. I also tested with 9.6 and there were no issues at all.

 I am aware of some internal changes in 9.7 in the `org-element` implementation that could affect the way of `poly-org` works. 

In the end, I uninstalled `poly-org` and everything works fine now. 

Just as reference, the bug is already reported upstream https://github.com/polymode/poly-org/issues/49

Thanks.
